### PR TITLE
resource/alicloud_kms_key: Added the field policy; Improved alicloud_kms_key testcase

### PR DESCRIPTION
**Pull request by MrWolong**

### CODE REVIEW

**Review by ChenHanZhang — COMMENTED** [review]
approved

File: alicloud/resource_alicloud_kms_key_test.go [hunk 1]
```diff
@@ -37,7 +37,7 @@ func init() {
 func testSweepKmsKey(region string) error {
```
Comment (MrWolong) [inline alicloud/resource_alicloud_kms_key_test.go:37]: TF_ACC=1 go test ./alicloud -v -run=TestAccAliCloudKmsKey_basic0 -timeout=1200m

=== RUN   TestAccAliCloudKmsKey_basic0

--- PASS: TestAccAliCloudKmsKey_basic0 (55.24s)
=== RUN   TestAccAliCloudKmsKey_basic0_twin

--- PASS: TestAccAliCloudKmsKey_basic0_twin (7.12s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	62.445s

TF_ACC=1 go test ./alicloud -v -run=TestAccAliCloudKmsKey_basic1 -timeout=1200m

=== RUN   TestAccAliCloudKmsKey_basic1

--- PASS: TestAccAliCloudKmsKey_basic1 (366.46s)
=== RUN   TestAccAliCloudKmsKey_basic1_twin

--- PASS: TestAccAliCloudKmsKey_basic1_twin (246.37s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	612.941s

TF_ACC=1 go test ./alicloud -v -run=TestAccAliCloudKmsKey_basic2 -timeout=[120](https://github.com/aliyun/terraform-provider-alicloud/actions/runs/9253772323/job/25454356588?pr=7264#step:7:121)0m

=== RUN   TestAccAliCloudKmsKey_basic2

--- PASS: TestAccAliCloudKmsKey_basic2 (56.10s)
=== RUN   TestAccAliCloudKmsKey_basic2_twin

--- PASS: TestAccAliCloudKmsKey_basic2_twin (7.59s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	63.781s

TF_ACC=1 go test ./alicloud -v -run=TestAccAliCloudKmsKey_basic3 -timeout=1200m

=== RUN   TestAccAliCloudKmsKey_basic3

--- PASS: TestAccAliCloudKmsKey_basic3 (54.53s)
=== RUN   TestAccAliCloudKmsKey_basic3_twin

--- PASS: TestAccAliCloudKmsKey_basic3_twin (7.58s)
PASS
ok  	github.com/aliyun/terraform-provider-alicloud/alicloud	62.186s

Finish!

File: alicloud/resource_alicloud_kms_key.go [hunk 2]
```diff
@@ -5,142 +5,156 @@ import (
...
+				Computed: true,
+			},
+			"delete_date": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"deletion_window_in_days": {
+				Type:          schema.TypeInt,
 				Optional:      true,
-				Computed:      true,
```
[the remark this answers is not in the document]
Comment (ChenHanZhang) [reply alicloud/resource_alicloud_kms_key.go:121]: 这两个字段如果是二选一，其中一个设置值的话，另一个未设置值也会读到值吧

File: alicloud/resource_alicloud_kms_key.go [hunk 3]
```diff
@@ -5,142 +5,156 @@ import (
...
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ValidateFunc:  StringInSlice([]string{"Enabled", "Disabled", "PendingDeletion"}, false),
+				ConflictsWith: []string{"key_state"},
 			},
 			"pending_window_in_days": {
 				Type:          schema.TypeInt,
 				Optional:      true,
-				Computed:      true,
```
[the remark this answers is not in the document]
Comment (MrWolong) [reply alicloud/resource_alicloud_kms_key.go:91]: pending_window_in_days只有在删除的时候才会被使用，查询接口无法返回，也没有set

File: alicloud/resource_alicloud_kms_key.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MrWolong) [reply alicloud/resource_alicloud_kms_key.go:121]: 不会，没有查询接口，没有set，所以这两个字段都没值

**Review by ChenHanZhang — COMMENTED** [review]
approved

**Review by ChenHanZhang — COMMENTED** [review]
approved

**Review by ChenHanZhang — COMMENTED** [review]
approved

**Pull request merged**